src/platform/plugins/shared/discover/public/application/main/discover_main_route.tsx

```diff
@@ -244,31 +242,48 @@ const DiscoverMainRouteContent = (props: SingleTabViewProps) => {
     );
   }
 
+  const tabsEnabled = discoverFeatureFlags.getTabsEnabled();
+  const shouldShowTabs = tabsEnabled && customizationContext.displayMode !== 'embedded';
+
   return (
     <rootProfileState.AppWrapper>
       <ChartPortalsRenderer runtimeStateManager={runtimeStateManager}>
-        <DiscoverTopNavMenuProvider>
-          <>
-            <h1 className="euiScreenReaderOnly" data-test-subj="discoverSavedSearchTitle">
-              {persistedDiscoverSession?.title
-                ? i18n.translate('discover.pageTitleWithSavedSearch', {
-                    defaultMessage: 'Discover - {savedSearchTitle}',
-                    values: {
-                      savedSearchTitle: persistedDiscoverSession.title,
-                    },
-                  })
-                : i18n.translate('discover.pageTitleWithoutSavedSearch', {
-                    defaultMessage: 'Discover - Session not yet saved',
-                  })}
-            </h1>
-            {tabsEnabled && customizationContext.displayMode !== 'embedded' ? (
-              <TabsView {...props} />
-            ) : (
-              <SingleTabView {...props} />
-            )}
```

Comment:
For the final version, we'll likely still need to conditionally render `TabsView` and `SingleTabView` to ensure the app menu renders when tabs are disabled too. I think we can keep what we had here mostly unchanged, and maybe just render the full version of the top nav above `SingleTabView`. I'm thinking something along these lines:
```tsx
const SingleTabViewWithTopNav = (props: SingleTabViewProps) => {
  const topNavMenu = useTopNavMenu();

  return (
    <>
      <TopNav topNavMenu={topNavMenu} />
      <SingleTabView {...props} />
    </>
  );
};

// ...

{tabsEnabled && customizationContext.displayMode !== 'embedded' ? (
  <TabsView {...props} />
) : (
  <SingleTabViewWithTopNav {...props} />
)}
```